plot.go

```diff
@@ -439,49 +439,71 @@ func (p *Plot) NominalY(names ...string) {
 	p.Y.Tick.Marker = ConstantTicks(ticks)
 }
 
-// Save saves the plot to an image file.  The file format is determined
-// by the extension.
+// WriterTo returns an io.WriterTo that will write the plot as
+// the specified image format.
 //
-// Supported extensions are:
+// Supported formats are:
 //
-//  .eps, .jpg, .jpeg, .pdf, .png, .svg, and .tiff.
-func (p *Plot) Save(w, h vg.Length, file string) (err error) {
+//  eps, jpg|jpeg, pdf, png, svg, and tif|tiff.
+func (p *Plot) WriterTo(w, h vg.Length, format string) (io.WriterTo, error) {
```

Comment:
Yeah. It was just a brainstorm, but basically it avoids an extra error check:
```wt, err := p.WriterTo(...)
if err != nil {
    whatever
}
if _, err := wt.WriteTo(w); err != nil {
   whatever else
}```

becomes what you wrote:
```
if _, err := p.WriterTo(...).WriteTo(w); err != nil {
    whatever
}
```